corehq/apps/ota/views.py

```diff
@@ -58,9 +59,11 @@ def get_restore_response(domain, couch_user, since=None, version='1.0',
                             status=401)
 
     project = Domain.get_by_name(domain)
+    app = Application.objects.get(app_id) if app_id else None
     restore_config = RestoreConfig(
         project=project,
         user=couch_user.to_casexml_user(),
+        app=app,
         params=RestoreParams(
```

Comment:
At first I thought it would be neater to pass the app_id in the GET params that the mobile sends instead of having a new URL. But putting it in the GET params from the mobile would involve changing the mobile code, and I thought it would be simpler to avoid that.

That choice affected my decision to pass `app` as a `RestoreConfig` parameter instead of as a `RestoreParams` parameter, because all the other `RestoreParams` values come from GET parameters from the mobile, and `app` or `app_id` would be the exception. I thought that could be confusing in the future, and less confusing to treat it the same way as `project`/`domain`, which also comes from the URL.

If you feel that it would be better to add it to `RestoreParams` to reduce the size of the diff, I'm happy to go with that. I don't feel strongly about this.